tycho-core/src/main/java/org/eclipse/tycho/core/osgitools/DependencyComputer.java

```diff
@@ -182,7 +187,7 @@ public List<DependencyEntry> computeDependencies(ModuleRevision module) {
         // sort by symbolicName_version to get a consistent order
         Map<String, BundleRevision> resolvedImportPackages = new TreeMap<>();
         for (BundleRevision bundle : visiblePackages.getParticipatingModules()) {
-            resolvedImportPackages.put(bundle.getSymbolicName(), bundle);
+            resolvedImportPackages.put(bundle.getSymbolicName() + "_" + bundle.getVersion(), bundle);
```

Comment:
Wouldn't it be simpler (and maybe faster) to do something like instead?
```
List<BundleRevision> bundles = new ArrayList<>(visiblePackages.getParticipatingModules());
bundles.sort(Comparator.comparing(Bundle::getSymbolicName).thenComparing(Bundle::getVersion));
```

Reply:
Or use a `TreeSet` if duplicates should be filtered.

Reply:
One don't even need to create a map / list at all I think and could merge it with the loop below, but to be honest I first want this ugly bug fixed before restructure the code to have a minimal change here.

Reply:
Understand. But just appending the version is not absolutly save.
Imaginge somebody has a Bundle `foo` with version `1` and another Bundle with name `foo_1` and version 2. Then this does not work.

Reply:
> Imagine somebody has a Bundle foo with version 1 and another Bundle with name foo_1 and version 2. Then this does not work.

Why? this will result in `foo_1`, and `foo_1_2` keys what are distinct strings :-)

Reply:
Hm. OK. It will probably work.